nix/modules/default/default.nix

```diff
@@ -0,0 +1,99 @@
+self: { config, lib, pkgs, spacebar, ... }:
+
+let
+  cfg = config.services.spacebarchat-server;
+  jsonFormat = pkgs.formats.json {};
+  configFile = jsonFormat.generate "spacebarchat-server.json" cfg.settings;
+in
+{
+  options.services.spacebarchat-server = {
+    enable = lib.mkEnableOption "spacebarchat-server";
+    package = lib.mkPackageOption self.packages.${pkgs.stdenv.hostPlatform.system} "spacebar-server" { default = "default"; };
+    extraEnvironment = lib.mkOption {
+      default = {};
+      description = ''
+        Environment variables passed to spacebarchat-server.
+        See <link xlink:href="https://docs.spacebar.chat/setup/server/configuration/env"/>.
+      '';
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf (oneOf [ str bool ]);
+        options = {
+          THREADS = lib.mkOption {
+            type = lib.types.ints.positive;
+            default = 1;
+            description = "Number of threads to run Spacebar on when using bundle. Make sure you've enabled RabbitMQ if using more than one.";
+          };
+          PORT = lib.mkOption {
+            type = lib.types.port;
+            default = 3001;
+            description = "Port to listen on. Used by all components, including bundle. If using bundle, all components run under the same port";
+          };
+          DATABASE = lib.mkOption {
+            type = lib.types.str;
+            default = "database.db";
+            example = "postgres://username:passwort@host-IP:port/databaseName";
+            description = "Database connection string. Defaults to SQLite3 at project root";
+          };
```

Comment:
Can we change this to a `lib.types.path` to avoid putting credentials in the nix store? I'd like to support this for all secrets in config aswell.